src/main/scala/de/retest/guistatemachine/api/impl/GuiStateMachineImpl.scala

```diff
@@ -3,33 +3,39 @@ package de.retest.guistatemachine.api.impl
 import com.typesafe.scalalogging.Logger
 import de.retest.guistatemachine.api.{GuiStateMachine, State, SutStateIdentifier}
 
-import scala.collection.concurrent.TrieMap
+import scala.collection.mutable.HashMap
 
 /**
   * Thread-safe implementation of a GUI state machine.
   */
 @SerialVersionUID(1L)
 class GuiStateMachineImpl extends GuiStateMachine with Serializable {
   @transient private val logger = Logger[GuiStateMachineImpl]
-  private var states = TrieMap[SutStateIdentifier, State]()
-
-  override def getState(sutState: SutStateIdentifier): State =
-    if (states.contains(sutState)) {
-      states(sutState)
-    } else {
-      logger.info(s"Create new state from SUT state $sutState")
-      val s = StateImpl(sutState)
-      states += (sutState -> s)
-      s
+  private var states = HashMap[SutStateIdentifier, State]()
+
+  override def createState(sutStateIdentifier: SutStateIdentifier, neverExploredActionTypesCounter: Int): State =
+    this.synchronized {
+      if (states.contains(sutStateIdentifier)) {
+        throw new RuntimeException(s"State from SUT state $sutStateIdentifier does already exist.")
```

Comment:
Hmmm, is the exception necessary? Can't we just return the existing state and log that it already exists?